docs/bricks_design.rst

```diff
@@ -0,0 +1,602 @@
+
+Bricks design explained
+=======================
+
+This is a runnable tutorial, which means you can copy its fragments 
+to an IPython notebook and run. The import section follows:
```

Comment:
Also copy-paste would still be there, even with IPython notebook. It is just that we have to show how the implementation evolves over time, and even though it is often changed only a little bit, I have to make a copy and edit it.